state/avro/src/main/java/com/expediagroup/streamplatform/streamregistry/state/avro/AvroObjectDeserializer.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (C) 2018-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.state.avro;
+
+import static com.fasterxml.jackson.core.JsonToken.END_ARRAY;
+import static com.fasterxml.jackson.core.JsonToken.END_OBJECT;
+import static com.fasterxml.jackson.core.JsonToken.START_OBJECT;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.function.BiConsumer;
+
+import lombok.extern.slf4j.Slf4j;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+
+@Slf4j
+class AvroObjectDeserializer extends JsonDeserializer<AvroObject> {
+  @Override
+  public AvroObject deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+    if (p.currentToken() != START_OBJECT) {
+      throw new IllegalStateException("Unexpected start token: " + p.currentToken());
```

Comment:
Won't deserialising do similar on failure?

Reply:
Not sure I follow the question

Reply:
If the first token isn't START_OBJECT I would expect deserialisation to fail anyway.
Not that important, just curious.

Reply:
Well you could do something silly like:

`mapper.convertValue(primitive, AvroObject.class)`

or

`mapper.convertValue(array, AvroObject.class)`